api/app/Http/Controllers/UserController.php

```diff
@@ -106,6 +107,33 @@ public function putOne($id, Request $request)
     }
 
     /**
+     * Patch an entity.
+     *
+     * @param  string   $id
+     * @param  Request  $request
+     * @return Response
+     */
+    public function patchOne($id, Request $request)
+    {
+        $this->validateId($id);
+        $this->repository->validateEmailConfirmationToken($request);
```

Comment:
Validation of a token shouldn't be a concern of a repository; it makes more sense to be a model method.